### ISSUE TITLE
Windows CUDA_PATH is set but CUDA wasn't able to be loaded

### ISSUE THREAD

**Issue by stentll** [microsoft/onnxruntime#11549]
**Describe the bug**
I am trying to create an inference session with CUDAExecutionProvider on Windows 10 but I get the following error when creating the session:

> 2022-05-17 13:32:04.3606784 [E:onnxruntime:Default, provider_bridge_ort.cc:995 onnxruntime::ProviderLibrary::Get] LoadLibrary failed with error 126 "The specified module could not be found." when trying to load "c:\Users\MyUserName\Miniconda3\envs\conda_env\lib\site-packages\onnxruntime\capi\onnxruntime_providers_cuda.dll"
> 
> RuntimeError: D:\a\_work\1\s\onnxruntime\python\onnxruntime_pybind_state.cc:531 onnxruntime::python::CreateExecutionProviderInstance CUDA_PATH is set but CUDA wasn't able to be loaded. Please install the correct version of CUDA and cuDNN as mentioned in the GPU requirements page (https://onnxruntime.ai/docs/reference/execution-providers/CUDA-ExecutionProvider.html#requirements), make sure they're in the PATH, and that your GPU is supported.

Note that the error starts with `D:\a\...` but I don't actually have a D:\ drive on my PC, only C:\

I have installed the necessary prerequisites (and correct versions of those) from ONNXRuntime site https://onnxruntime.ai/docs/execution-providers/CUDA-ExecutionProvider.html.

My system variables include:
CUDA_PATH=C:\Program Files\NVIDIA GPU Computing Toolkit\CUDA\v11.4
CUDA_PATH_V11_4=C:\Program Files\NVIDIA GPU Computing Toolkit\CUDA\v11.4
cuDNN_PATH=C:\Program Files\NVIDIA\CUDNN\v8.2.2.26

and also system PATH variables:
C:\Program Files\NVIDIA GPU Computing Toolkit\CUDA\v11.4\bin
C:\Program Files\NVIDIA GPU Computing Toolkit\CUDA\v11.4\libnvvp
C:\Program Files\NVIDIA\CUDNN\v8.2.2.26\bin
C:\Program Files\NVIDIA\CUDNN\v8.2.2.26\lib\x64

I feel like I have completed the advice in the error message - CUDA path set, correct versions of CUDA and cuDNN installed, GPU supported - but I still can't use the EP. What am I missing?

**Urgency**
I would like to solve this by end of May 2022.

**System information**
- OS Platform and Distribution: Windows 10 Enterprise 20H2, os build 19042.1645
- ONNX Runtime installed from: pip install onnxruntime-gpu==1.10
- ONNX Runtime version: 1.10.0
- Python version: 3.8.13
- CUDA/cuDNN version: CUDA==11.4, cuDNN==8.2.2.26
- GPU model and memory: NVIDIA RTX T1200, 4096MiB, Turing architecture (more info at https://www.notebookcheck.net/NVIDIA-T1200-Laptop-GPU-GPU-Benchmarks-and-Specs.532550.0.html)

**To Reproduce**
Run the following piece of code with any .onnx model.
```python
import onnxruntime as ort

providers = [
    (
        "CUDAExecutionProvider",
        {
            "device_id": 0,
            "arena_extend_strategy": "kSameAsRequested",
            "cudnn_conv_algo_search": "HEURISTIC",
            "do_copy_in_default_stream": True,
        },
    ),
    "CPUExecutionProvider",
]
sess = ort.InferenceSession(my_model_path, providers=providers)
```

**Expected behavior**
I would expect the session creation to succeed and model being eventually run on GPU. This works well on my Ubuntu setup but I have not so far succeeded on getting it working on a Windows system.

**Comment by yuslepukhin** [comment]
The `D:\` part just points to the source file on a build box, so it had nothing to do with your machine. You might want to check if those CUDA folders are present in the `PATH` of your Conda environment. Easy way to check is to use `where` command in CMD.

**Comment by skottmckay** [comment]
Did you install the zlib dll and is that accessible via an entry in the PATH?

**Comment by duli2012** [comment]
Did you add the path to the cuDNN bin directory to the PATH environment variable?

**Comment by stentll** [comment]
First of all, thank you all for your rapid responses

Your ideas lead me to fix the issue. @yuslepukhin when you mentioned that the PATH might not be visible inside my conda env then I decided to try what happens if I instead will use a virtualenv environment. I created one, installed all dependencies and it worked. So the issue seemed to be indeed with the conda environment or how I was using it.

In windows I had problems using my conda env in VSCode. I was running all my scripts in VSCode terminal. Once I started a file using the conda env I got "ImportError: DLL load failed while importing _ssl: The specified module could not be found" which lead me to a fix on the web to circumvent this:
1. open Anaconda prompt
2. open vscode with `code .`
This way conda env got 'properly' loaded. But it also seems that this is what caused the PATH issues with onnxruntime.

I went back to the original error of DLL loading failed and fixed this using advice I got from here: https://stackoverflow.com/questions/54175042/python-3-7-anaconda-environment-import-ssl-dll-load-fail-error/60405693#60405693 
One specific comment under the answer helped my case: "I've added the dll in the DLLs folder inside my environment folder (not Anaconda3 one) and it seemed to work. Thanks – 
SJGD Feb 19, 2021 at 4:30"

So once the DLL error was fixed I opened VSCode on windows like a normal person (without using Anaconda prompt) and started everything up and it worked!

When I started the test script straight from Anaconda prompt then it worked as well. So prompt+VSCode combination was not working well.


Thanks again for your help. This issue is solved for me now.
